### DISCUSSION TITLE
/nicklist to show all nicknamed players

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#7347] [Enhancement]
Keep track of all nicknamed players with `/nicklist` quite important if you have a lot of players.

Should display their current/past usernames alongside their nick and the date it was set.

Same dispaly as everything else in CMI text, Press the X to remove their nickname, order by range of closes player with a nickname, or fallback to last nickname set based on date.

Clicking the Text would let you edit their nickname, it would pre-fill your chat with `/nick {username} CurrentNick` 

```
[X] 01. Nickname (Username) Date Set
```

**Comment by MrCrazys** [comment]
> Keep track of all nicknamed players with `/nicklist` quite important if you have a lot of players.
> 
> Should display their current/past usernames alongside their nick and the date it was set.
> 
> Same dispaly as everything else in CMI text, Press the X to remove their nickname, order by range of closes player with a nickname, or fallback to last nickname set based on date.
> 
> Clicking the Text would let you edit their nickname, it would pre-fill your chat with `/nick {username} CurrentNick`
> 
> ```
> [X] 01. Nickname (Username) Date Set
> ```

I like the idea of `/cmi nicklist`, but I don't know how easy it would be to keep track of all past nicknames. Keeping track of this data would probably be very taxing, and would likely fill up the database very quickly. If a player was to keep changing their nickname, it would have to keep logging each nickname to the database. I suppose what we could have to mitigate this is a clear date, where it will clear nicknames from the database after a certain amount of time in days. It would definitely be easier than having to use a Database Browser to remove past nicknames.

**Reply by TomLewis** [reply]
> Keeping track of this data would probably be very taxing, and would likely fill up the database very quickly.

Nah, you just make a new table haha `CMI_NicknameHistory`, Schema only needs to be simple. `ID` `UID` `nickname` `timestamp` 

On my server, players cant set their own nickname, only admins can, so I wouldnt have a massive database, but even with a million rows in MySQL thats nothing at all in terms of space.

**Reply by MrCrazys** [reply]
In your case, yes, but for servers that allow players to change their nickname, the database would likely fill up very quickly.

**Reply by TomLewis** [reply]
Wouldnt matter if its own table, have a billion rows.

**Comment by TomLewis** [comment]
Bump
